apis/v1alpha1/httproute_types.go

```diff
@@ -328,79 +328,94 @@ type HTTPRouteMatch struct {
-	//   conformance rings. Filter-specific configuration for such filters
+	//   conformance levels. Filter-specific configuration for such filters
 	//   is specified using the ExtensionRef field. `Type` should be set to
-	//   "ImplementationSpecific" for custom filters.
+	//   "Custom" for custom filters.
 	//
-	// Implementers are encouraged to define custom implementation
-	// types to extend the core API with implementation-specific behavior.
+	// Implementers are encouraged to define custom implementation types to
+	// extend the core API with implementation-specific behavior.
 	//
 	// +unionDiscriminator
-	// +kubebuilder:validation:MinLength=1
-	// +kubebuilder:validation:MaxLength=100
-	Type string `json:"type"`
+	Type HTTPRouteFilterType `json:"type,omitempty"`
 
-	// ExtensionRef is an optional, implementation-specific extension to the
-	// "filter" behavior.  The resource may be "configmap" (use the empty
-	// string for the group) or an implementation-defined resource (for
-	// example, resource "myroutefilters" in group "networking.acme.io").
-	// Omitting or specifying the empty string for both the resource and
-	// group indicates that the resource is "configmaps".
-	// ExtensionRef MUST NOT be used for core and extended filters.
+	// RequestHeaderModifier defines a schema for a filter that modifies request
+	// headers.
+	//
+	// Support: Core
+	//
 	// +optional
-	ExtensionRef *LocalObjectReference `json:"extensionRef,omitempty"`
-
-	// Filter-specific configuration definitions for core and extended filters
+	RequestHeaderModifier *HTTPRequestHeaderFilter `json:"requestHeaderModifier,omitempty"`
 
+	// RequestMirror defines a schema for a filter that mirrors requests.
+	//
+	// Support: Extended
+	//
 	// +optional
-	RequestHeader *HTTPRequestHeaderFilter `json:"requestHeader,omitempty"`
+	RequestMirror *HTTPRequestMirrorFilter `json:"requestMirror,omitempty"`
 
+	// ExtensionRef is an optional, implementation-specific extension to the
+	// "filter" behavior.  The resource may be "configmap" or an
+	// implementation-defined resource (for example, resource "myroutefilters"
+	// in group "networking.acme.io"). ExtensionRef MUST NOT be used for core
+	// and extended filters.
+	//
+	// Support: Implementation-specific
+	//
 	// +optional
-	RequestMirror *HTTPRequestMirrorFilter `json:"requestMirror,omitempty"`
+	ExtensionRef *LocalObjectReference `json:"extensionRef,omitempty"`
 }
 
-// HTTPRequestHeaderFilter defines configuration for the
-// RequestHeader filter.
+// HTTPRouteFilterType identifies a type of HTTPRoute filter.
+// +kubebuilder:validation:Enum=RequestHeaderModifier;RequestMirror;Custom
+type HTTPRouteFilterType string
+
+const (
+	// RequestHeaderModifierHTTPRouteFilter can be used to add or remove an HTTP
+	// header from an HTTP request before it is sent to the upstream target.
+	//
+	// Support: Core
+	RequestHeaderModifierHTTPRouteFilter HTTPRouteFilterType = "RequestHeaderModifier"
```

Comment:
In most places in this API, each `FooType` resource is named as `FooBar`.

Should we name this as `HTTPRouteFilterRequestHeaderModifier`?

This has a benefit of grouping the related constants in Godoc and also helps with code-completion like tools.